ui/app/pages/settings/networks-tab/networks-tab.component.js

```diff
@@ -45,7 +55,7 @@ export default class NetworksTab extends PureComponent {
           !this.isCurrentPath(SETTINGS_ROUTE) && (
             <div
               className="settings-page__back-button"
-              onClick={networkIsSelected || networksTabIsInAddMode
+              onClick={(networkIsSelected && !networkDefaultedToProvider) || networksTabIsInAddMode
```

Comment:
Not for this PR - we should refactor how routing works in setting view should we dont have to cross check `state` and `router` all the time. Ideally, we should be able to just call `this.props.history.goBack()` here

Reply:
These concerns should be removed with 6ddb24ff26f54466880295b381ad46dc975642f5